### PR TITLE
Fixed exception: "InvalidOperationException: Source column 'Label' is required but not found."

### PR DESCRIPTION
**Pull request by zeahmed**
This PR fixes issue #117 

Added a test to cover the issue. Please see issue #117 for more details.

### CODE REVIEW

File: src/Microsoft.ML.Api/SchemaDefinition.cs [hunk 1]
```diff
@@ -332,8 +332,18 @@ public static SchemaDefinition Create(Type userType)
...
+                var mappingAttr = fieldInfo.GetCustomAttribute<ColumnAttribute>();
+                string attrName = null;
+                if (mappingAttr != null)
+                {
+                    attrName = mappingAttr.Name;
+                }
+                else
+                { 
+                    var mappingNameAttr = fieldInfo.GetCustomAttribute<ColumnNameAttribute>();
+                    attrName = mappingNameAttr != null ? mappingNameAttr.Name : null;
```
Comment (TomFinley) [inline src/Microsoft.ML.Api/SchemaDefinition.cs]: >mappingNameAttr != null ? mappingNameAttr.Name : null; [](start = 31, length = 54)

This line is more clearly and concisely expressed as `attrName = mappingNameAttr?.Name;`

File: src/Microsoft.ML.Api/SchemaDefinition.cs [hunk 2]
```diff
@@ -332,8 +332,18 @@ public static SchemaDefinition Create(Type userType)
...
+                }
+                var name = attrName == null ? fieldInfo.Name : attrName;
```
Comment (TomFinley) [inline src/Microsoft.ML.Api/SchemaDefinition.cs]: >var name = attrName == null ? fieldInfo.Name : attrName; [](start = 16, length = 56)

More clearly and concisely as `var name = attrName ?? fieldInfo.Name`.

File: src/Microsoft.ML.Api/SchemaDefinition.cs [hunk 3]
```diff
@@ -332,8 +332,18 @@ public static SchemaDefinition Create(Type userType)
...
                 // Disallow duplicate names, because the field enumeration order is not actually
```
Comment (TomFinley) [inline src/Microsoft.ML.Api/SchemaDefinition.cs]: Actually wait... if you just cut out these 12 lines of code and replaced it with these:

```csharp
var mappingAttr = fieldInfo.GetCustomAttribute<ColumnAttribute>();
var mappingNameAttr = fieldInfo.GetCustomAttribute<ColumnNameAttribute>();
string name = mappingAttr?.Name ?? mappingNameAttr?.Name ?? fieldInfo.Name;
```

you could make this much shorter, and much, much clearer. The `??` and `?.` operators are fantastic aids for this sort of "if null/not null" logic...

**Comment by glebuk** [comment]
@glebuk is added to the review. #Closed

**Comment by zeahmed** [comment]
Thanks for the review.

**Pull request merged**